src/index.ts

```diff
@@ -23,7 +23,7 @@ import flatten, { unflatten } from 'flat'
 type CartesianData<T> = { [P in keyof T]: Array<T[P]> | any }
 
 const xproduct = (vals: any[][]) =>
-  reduce((a: any[][], b: any[]) => flatMap(x => map(y => concat(x, y))(b))(a))([
+  reduce((a: any[][], b: any[]) => flatMap(x => map(y => concat(x, [y]))(b))(a))([
```

Comment:
This is to support values that are arrays. Before this change xproduct wouldn't process this correctly:
`xproduct([[1,2,3], [1,2, ['a','b','c']]])` because the second value of the second array is an array itself.